File: 3UI/13LessonUITabBarController/LessonUITableBarController/SecondViewController.m
//
//  SecondViewController.m
//  LessonUITableBarController
//
//  Created by lanouhn on 15/11/18.
//  Copyright (c) 2015年 lanouhn0. All rights reserved.
//

#import "SecondViewController.h"

@interface SecondViewController ()

@end

@implementation SecondViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    self.view.backgroundColor = [UIColor orangeColor];
}

- (void)viewWillAppear:(BOOL)animated {
    NSLog(@"第二个界面即将出现");
}

- (void)viewDidAppear:(BOOL)animated {
    NSLog(@"第二个界面已经出现");
}

- (void)viewWillDisappear:(BOOL)animated {
    NSLog(@"第二个界面即将消失");
}

- (void)viewDidDisappear:(BOOL)animated {
    NSLog(@"第二个界面已经消失");
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: 5Swift/16 Deinitialization/Deinitialization/SecondViewController.swift
//
//  SecondViewController.swift
//  Deinitialization
//
//  Created by admin on 2018/3/20.
//  Copyright © 2018年 admin. All rights reserved.
//

import UIKit

class SecondViewController: UIViewController {

    var arr = [""]
    
    override func viewDidLoad() {
        super.viewDidLoad()

    }

    deinit {
        print("***********")
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    

    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}
